### PR TITLE
sceNp: sceNpMatching/Commerce error checks

### PR DESCRIPTION
**Pull request by Megamouse**
- Reverse engineered the error checks of sceNpMatching (waiting for RPCN 1.0 for cleanup)

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 1]
```diff
@@ -3748,88 +4140,487 @@ error_code sceNpManagerSubSignout(vm::ptr<SceNpId> npId)
...
 {
-	sceNp.todo(
-	    "sceNpMatchingGetRoomListLimitGUI(ctx_id=%d, communicationId=*0x%x, range=*0x%x, cond=*0x%x, attr=*0x%x, handler=*0x%x, arg=*0x%x)", ctx_id, communicationId, range, cond, attr, handler, arg);
+	sceNp.todo("sceNpMatchingGetRoomListLimitGUI(ctx_id=%d, communicationId=*0x%x, range=*0x%x, cond=*0x%x, attr=*0x%x, handler=*0x%x, arg=*0x%x)", ctx_id, communicationId, range, cond, attr, handler, arg);
+
+	if (false) // TODO
+		return SCE_NP_MATCHING_ERROR_UTILITY_UNAVAILABLE;
+
+	// This check is used in all sceNpMatchingGetRoomList functions
+	error_code err = check_room_list_params(ctx_id, communicationId, range, cond, attr, handler, 1, 0);
```
Comment (RipleyTom) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp:4610]: Any purpose to keeping the last 2 params if they're fixed?

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 2]
```diff
@@ -5723,15 +6850,39 @@ error_code sceNpSignalingGetPeerNetInfoResult(u32 ctx_id, u32 req_id, vm::ptr<Sc
...
+	if (!npId)
+		return SCE_NP_UTIL_ERROR_INVALID_ARGUMENT;
+
+	if (npId->reserved[0] != 1)
+		return SCE_NP_UTIL_ERROR_INVALID_NP_ID;
+
+	if (!npId->unk1[1])
+	{
+		npId->unk1[1] = "ps3\0"_u32;
+	}
```
Comment (RipleyTom) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp:6866]: Please comment all this for now.

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 3]
```diff
@@ -5723,15 +6850,39 @@ error_code sceNpSignalingGetPeerNetInfoResult(u32 ctx_id, u32 req_id, vm::ptr<Sc
...
+	if (!npId)
+		return SCE_NP_UTIL_ERROR_INVALID_ARGUMENT;
+
+	if (npId->reserved[0] != 1)
+		return SCE_NP_UTIL_ERROR_INVALID_NP_ID;
+
+	if (!npId->unk1[1])
+	{
+		npId->unk1[1] = "psp\0"_u32; // TODO: confirm
+	}
```
Comment (RipleyTom) [inline rpcs3/Emu/Cell/Modules/sceNp.cpp:6884]: Please comment all this for now.

File: rpcs3/Emu/Cell/Modules/sceNp.cpp [hunk 4]
```diff
@@ -2194,214 +2271,492 @@ error_code sceNpBasicGetEvent(vm::ptr<s32> event, vm::ptr<SceNpUserInfo> from, v
...
+		return SCE_NP_COMMERCE_ERROR_CTX_NOT_FOUND;
+
 	return CELL_OK;
 }
 
 error_code sceNpCommerceInitProductCategory(vm::ptr<SceNpCommerceProductCategory> pc, vm::cptr<void> data, u32 data_size)
 {
 	sceNp.todo("sceNpCommerceInitProductCategory(pc=*0x%x, data=*0x%x, data_size=%d)", pc, data, data_size);
+
+	if (!pc) // Not really checked
```
[the remark this answers is not in the document]
Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/sceNp.cpp:2304]: Idk. feels like this would help no-one

File: rpcs3/Emu/Cell/Modules/sceNp.cpp (hunk 1 above)
Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/sceNp.cpp:4610]: check_room_list_params is also used in sceNpMatchingGetRoomList with at least one different parameter

**Pull request merged**